sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineRunner.java

```diff
@@ -420,6 +426,37 @@ public static DataflowPipelineRunner fromOptions(PipelineOptions options) {
     return super.apply(new AssignWindows<>(transform), input);
   }
 
+  @Nullable
+  private void maybeRegisterDebuggee(DataflowPipelineOptions options, String uniquifier) {
```

Comment:
Done. Given the dependency on pipeline options to create the client, etc. I'm not sure how reusable this will be, but I do think it is clearer to separate the options interactions from the debugger interactions.